src/main/thrift/parquet.thrift

```diff
@@ -576,5 +631,8 @@ struct FileMetaData {
    * e.g. impala version 1.0 (build 6cf94d29b2b7115df4de2c06e2ab4326d721eb55)
    **/
   6: optional string created_by
+
+  /** Sort order used for each column in this file */
+  7: optional list<ColumnOrder> column_orders;
```

Comment:
Another option is to add an optional column_order field in SchemaElement which would set only for primitive types.
``` 10: optional ColumnOrder column_order; ```
https://github.com/rdblue/parquet-format/blob/9962df8e0ea85858cf032451d0ee83ec3f4d39fe/src/main/thrift/parquet.thrift#L260